backend/tests/unit/dal/Services/Admin/ProvinceServiceTest.cs

```diff
@@ -0,0 +1,144 @@
+            using var init = helper.InitializeDatabase(user);
+
+            var service = helper.CreateService<ProvinceService>(user);
+
+            // Act
+            var result = service.Get();
+
+            // Assert
+            Assert.True(result.Count() > 0);
+        }
+        #endregion
+        #region Update
+        [Fact]
+        public void Update_Province()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.SystemAdmin);
+            var province = EntityHelper.CreateProvince("TP", "Test Province");
+            helper.CreatePimsContext(user, true).AddAndSaveChanges(province);
+
+            var service = helper.CreateService<ProvinceService>(user);
+
+            var updateProvince = EntityHelper.CreateProvince("TP", "Test Province");
+
+            var newName = "Updated Province";
+            updateProvince.Name = newName;
+
+            // Act
+            service.Update(updateProvince);
+
+            // Assert
+            province.Name.Should().Be(newName);
+        }
+
+        [Fact]
+        public void Update_Province_ArgumentNullException()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission();
+            var service = helper.CreateService<ProvinceService>(user);
+
+            // Act
+            // Assert
+            Assert.Throws<ArgumentNullException>(() =>
+                service.Update(null));
+        }
+
+        [Fact]
+        public void Update_Province_KeyNotFound()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission();
+            var province = EntityHelper.CreateProvince("TP", "Test Province");
+
+            helper.CreatePimsContext(user, true);
+
+            var service = helper.CreateService<ProvinceService>(user);
+            var context = helper.GetService<PimsContext>();
+
+            // Act
+            // Assert
+            Assert.Throws<KeyNotFoundException>(() =>
+                service.Update(province));
+        }
+        #endregion
+        #region Remove
+        [Fact]
+        public void Remove_Province()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.SystemAdmin, Permissions.AdminRoles);
+            var province = EntityHelper.CreateProvince("TP", "Test Province");
+
+            helper.CreatePimsContext(user, true).AddAndSaveChanges(province);
+
+            var service = helper.CreateService<ProvinceService>(user);
+            var context = helper.GetService<PimsContext>();
+
+            // Act
+            service.Remove(province);
+
+            // Assert
+            Assert.Equal(EntityState.Detached, context.Entry(province).State);
```

Comment:
This kind of assert could be applied to other add/update tests to make them more robust.

Reply:
Gotchya, thanks! Will adjust the other tests. Will also fix the `result.Count()` asserts.